connectors/src/connectors/slack/temporal/activities.ts

```diff
@@ -493,33 +494,51 @@ export async function syncThread(
   let next_cursor = undefined;
 
   do {
-    const replies: ConversationsRepliesResponse =
-      await client.conversations.replies({
-        channel: channelId,
-        ts: threadTs,
-        cursor: next_cursor,
-        limit: 100,
-      });
-    // Despite the typing, in practice `replies` can be undefined at times.
-    if (!replies) {
-      const workflowError: WorkflowError = {
-        type: "transient_upstream_activity_error",
-        message:
-          "Received unexpected undefined replies from Slack API in syncThread (generally transient)",
-        __is_dust_error: true,
-      };
-      throw workflowError;
-    }
-    if (replies.error) {
-      throw new Error(replies.error);
-    }
-    if (!replies.messages) {
-      break;
+    try {
+      const replies: ConversationsRepliesResponse =
+        await client.conversations.replies({
+          channel: channelId,
+          ts: threadTs,
+          cursor: next_cursor,
+          limit: 100,
+        });
+      // Despite the typing, in practice `replies` can be undefined at times.
+      if (!replies) {
+        const workflowError: WorkflowError = {
+          type: "transient_upstream_activity_error",
+          message:
+            "Received unexpected undefined replies from Slack API in syncThread (generally transient)",
+          __is_dust_error: true,
+        };
+        throw workflowError;
+      }
+      if (replies.error) {
+        throw new Error(replies.error);
+      }
+      if (!replies.messages) {
+        break;
+      }
+      allMessages = allMessages.concat(
+        replies.messages.filter((m) => !!m.user)
+      );
+      next_cursor = replies.response_metadata?.next_cursor;
+    } catch (e) {
+      const slackError = e as CodedError;
+      if (slackError.code === ErrorCode.PlatformError) {
+        const platformError = slackError as WebAPIPlatformError;
+        if (platformError.data.error === "channel_not_found") {
```

Comment:
wasn't the error `thread_not_found` instead of `channel_not_found` ?

Reply:
ah yeah fixing

Reply:
thanks for catching